src/unix/core.c

```diff
@@ -1041,6 +1043,48 @@ int uv_getrusage(uv_rusage_t* rusage) {
 }
 
 
+int uv_getrusage(uv_rusage_t* rusage) {
+  return uv__getrusage(RUSAGE_SELF, rusage);
+}
+
+
+int uv_getrusage_thread(uv_rusage_t* rusage) {
+#if defined(__APPLE__)
+  mach_msg_type_number_t count;
+  thread_basic_info_data_t info;
+  kern_return_t kr;
+  thread_t thread;
+
+  thread = mach_thread_self();
+  count = THREAD_BASIC_INFO_COUNT;
+  kr = thread_info(thread,
+                   THREAD_BASIC_INFO,
+                   (thread_info_t)&info,
+                   &count);
+
+  if (kr != KERN_SUCCESS) {
+    mach_port_deallocate(mach_task_self(), thread);
+    return UV_EINVAL;
+  }
+
+  memset(rusage, 0, sizeof(*rusage));
+
+  rusage->ru_utime.tv_sec = info.user_time.seconds;
+  rusage->ru_utime.tv_usec = info.user_time.microseconds;
+  rusage->ru_stime.tv_sec = info.system_time.seconds;
+  rusage->ru_stime.tv_usec = info.system_time.microseconds;
+
+  mach_port_deallocate(mach_task_self(), thread);
```

Comment:
Just a minor DRY suggestion:
```suggestion
  kr = thread_info(thread,
                   THREAD_BASIC_INFO,
                   (thread_info_t)&info,
                   &count);
  mach_port_deallocate(mach_task_self(), thread);
  if (kr != KERN_SUCCESS) {
    return UV_EINVAL;
  }

  memset(rusage, 0, sizeof(*rusage));
  rusage->ru_utime.tv_sec = info.user_time.seconds;
  rusage->ru_utime.tv_usec = info.user_time.microseconds;
  rusage->ru_stime.tv_sec = info.system_time.seconds;
  rusage->ru_stime.tv_usec = info.system_time.microseconds;
```

Reply:
Oh, true that! Thanks for the catch! It's fixed now!